### PR TITLE
Vending machine ui

### PR DESCRIPTION
**Pull request by Alex-Selvey**
Adapted coding from oracle pull request #50
shamblers

shadycigs

🆑 MemeProof
add: Adds a nicer UI for vending machines.
tweak: Vending machines put items into your hand if they can.
ongoing: still working on migrating clothes
/🆑

### CODE REVIEW

**Comment by ghost** [comment]
Somehow you messed up the commits? The oracle PDA UI PR seems to have been re-applied despite having been merged????

**Comment by alexkar598** [comment]
Use git rebase instead of git merge

**Comment by Alex-Selvey** [comment]
I'll just start putting everything under its own branch. I figured this was a small update but if it's an issue I'll file other updates under an improvements branch

**Review by ghost — CHANGES_REQUESTED** [review]
No need to mirror the entire file. You only need to mirror/override `ui_interact`

Also, update it to work with economy

**Comment by Alex-Selvey** [comment]
Alright no problem. I'll work on it tonight after I get off from work.

File: sunset/code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+	popup.set_title_image(user.browse_rsc_icon(icon, icon_state))
+	popup.open()
+
+GLOBAL_LIST_EMPTY(vending_machine_icon)
+/obj/machinery/vending/proc/GetIconForProduct(datum/data/vending_product/P)
+	var/producticon = GLOB.vending_machine_icon[P.product_path]
+	if (producticon)
+		return producticon
+	var/product = new P.product_path()
+	producticon = icon2base64html(product)
```
Comment (FlattestGuitar) [inline sunset/code/modules/vending/_vending.dm]: you should use `icon2base64(getFlatIcon(product, no_anim = TRUE))` to cover items with overlays, as well as animated icons

**Comment by Alex-Selvey** [comment]
Updated to master to continue updating. Still working on changes.

File: sunset/code/modules/vending/_vending.dm (hunk 1 above)
Comment (Alex-Selvey) [reply sunset/code/modules/vending/_vending.dm]: I'm still working on it, but this is what happens if I just swap that out:
change of geticon

Comment (FlattestGuitar) [reply sunset/code/modules/vending/_vending.dm]: also this: ``

**Comment by Alex-Selvey** [comment]
FlattestGuitar I got that code working like you suggested.

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -468,7 +468,12 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 		use_power(5)
 		if(icon_vend) //Show the vending animation if needed
 			flick(icon_vend,src)
-		new R.product_path(get_turf(src))
+		var/vended = new R.product_path(get_turf(src))
```
Comment (ghost) [inline code/modules/vending/_vending.dm:472]: `// sunset start`

File: code/modules/vending/_vending.dm [hunk 3]
```diff
@@ -468,7 +468,12 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
...
+		if(usr.can_put_in_hands(vended))
+			usr.put_in_hands(vended)
+			to_chat(usr, "<span class='notice'>You take the [R.name] out of the slot.</span>")
+		else
+			to_chat(usr, "<span class='warning'>The [R.name] falls onto the floor!</span>")
```
Comment (ghost) [inline code/modules/vending/_vending.dm:477]: `// sunset end

File: sunsetstation.dme [hunk 4]
```diff
@@ -2708,6 +2708,7 @@
 #include "code\modules\vehicles\vehicle_key.dm"
 #include "code\modules\vehicles\cars\car.dm"
 #include "code\modules\vehicles\cars\clowncar.dm"
+#include "code\modules\vending\_vending.dm"
 #include "sunset\code\modules\vending\_vending.dm"
```
Comment (ghost) [inline sunsetstation.dme]: Don't edit the DME manually!!!!!!!

**Review by ghost — CHANGES_REQUESTED** [review]

File: sunsetstation.dme (hunk 4 above)
Comment (Alex-Selvey) [reply sunsetstation.dme]: I had accidentally removed the original line for _vending so I just put it back so it looks like that in the commit history but was just a mistake.

File: code/modules/vending/_vending.dm (hunk 2 above)
Comment (Alex-Selvey) [reply code/modules/vending/_vending.dm:472]: done

File: code/modules/vending/_vending.dm (hunk 3 above)
Comment (Alex-Selvey) [reply code/modules/vending/_vending.dm:477]: done

File: sunsetstation.dme (hunk 4 above)
Comment (ghost) [reply sunsetstation.dme]: I'll merge this if you fix it. All you need to do to re-sort DME is to just click compile in Dream Maker. You don't even need to wait for it to compile. You can just compile, then immediately stop the compilation.

File: code/modules/vending/_vending.dm (hunk 3 above)
Comment (MemeProof) [reply code/modules/vending/_vending.dm]: Resorted DME, thank you for the explanation.

File: sunsetstation.dme (hunk 4 above)
Comment (Alex-Selvey) [reply sunsetstation.dme]: Re-sorted the DME and thank you for the explanation and I'll make sure it's sorted in the future.

**Review by ghost — APPROVED** [review]
Looks good to me. Did you test it in game?

**Comment by Alex-Selvey** [comment]
I did

**Comment by ghost** [comment]
Ok, I'll merge when travis compiles.

**Pull request merged**